sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -528,6 +529,7 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    * <li>`header` (default `false`): uses the first line as names of columns.</li>
    * <li>`inferSchema` (default `false`): infers the input schema automatically from data. It
    * requires one extra pass over the data.</li>
+   * <li>`samplingRatio` (default 1.0): the sample ratio of rows used for schema inferring.</li>
```

Comment:
Also need to update the PySpark API

Reply:
I just did absolutely the same as `samplingRatio` for JSON datasource - `samplingRatio` is not supported as a parameter of `json()` but could be set as the option like  `.option('samplingRatio', '0.1')`.  Should I update PySpark API regarding `samplingRatio` for `json()` in separate PR?

Reply:
Yes, please. Seems this option can be useful given your investigation and @rxin's opinion.

Reply:
Please, have a look at the PR: https://github.com/apache/spark/pull/21056